src/Native/System.Private.CoreLib.Native/pal_guid.cpp

```diff
@@ -20,13 +20,18 @@ typedef struct _GUID {
 extern "C" void CoreLibNative_CreateGuid(GUID* pGuid)
 {
 #if HAVE_LIBUUID_H
+#ifdef _WASM_
```

Comment:
It would be nice to do this using `HAVE_XXX` autogenerated defines for portability, not hard coding it to WASM.

Reply:
Will do.

My only concern here would be that `uuid_generate_random` will always provide version 4 UUIDs, where as `uuid_generate` will fall back to version 1 UUIDs, based on current time and MAC address. Where as we know Emscripten implements `uuid_generate` using version 4 UUIDs, this could result in information leakage on other platforms in the future.